qiskit_machine_learning/kernels/algorithms/quantum_kernel_trainer.py

```diff
@@ -22,7 +23,11 @@
 from qiskit.algorithms.variational_algorithm import VariationalResult
 from qiskit_machine_learning.utils.loss_functions import KernelLoss, SVCLoss
 
-from qiskit_machine_learning.kernels import QuantumKernel
+from qiskit_machine_learning.kernels import QuantumKernel as QuantumKernelOld
+from qiskit_machine_learning.kernels import TrainableFidelityQuantumKernel
+
+
+QuantumKernel = Union[QuantumKernelOld, TrainableFidelityQuantumKernel]
```

Comment:
I highly doubt that this would be straightforward. The whole reason for this PR (before we started adding the primitives) was that it was basically impossible to extend the QuantumKernel class in a clean way.

Reply:
I was not looking to extend the old QuantumKernel. More would it be simple enough to adapt it a little so the old code looked like the new way wrt to the interface e.g. BaseKernel that I commented elsewhere or TrainableKernelMixin here. The intent remains to drop it going forwards I was more wondering if adapting it lightly would mean this code could be what we want going forwards at this point rather than having some Union and having to change this once again when the QuantumKernel gets dropped.

Reply:
Yeah, no more unions.